lib/workers/repository/updates/generate.spec.ts

```diff
@@ -125,14 +138,14 @@ describe('workers/repository/updates/generate', () => {
             foo: 2,
           },
         },
-      ]);
+      ] satisfies BranchUpgradeConfig[];
       const res = generateBranchConfig(branch);
       expect(res.foo).toBe(1);
       expect(res.groupName).toBeUndefined();
     });
 
     it('groups multiple upgrades same version', () => {
-      const branch = partial<BranchUpgradeConfig>([
+      const branch: BranchUpgradeConfig[] = [
```

Comment:
This one needs some digging into, because it cause error when only satisfies is used.
I think the error is caused because we have 3 diff values for `constraints` field, 
1. field not present
2. field has 2 values `foo`, `bar` both having string values
3. field only has `foo` with string value

In `BranchUpgradeConfig` the defn for `constraints` is 
`constraints?:Record<string, string>`

By this def all 3 values above are correct. But TS throws this error
```log
Types of property 'constraints' are incompatible.
        Type '{ foo: string; bar?: undefined; }' is not assignable to type 'Record<string, string>'.
          Property 'bar' is incompatible with index signature.
            Type 'undefined' is not assignable to type 'string'.
```

Reply:
My guess is that this happens because `satisfies` doesn't assert a type on the config-objject rather just checks if it is valid.
Thus when the upgrade-array(_branch_) is passed to the function it tries to make sense by merging the type of all the objects in the array, however they don't have their own strict type so the confusion.

I think for that it is okay for us to assert the type once we have validated the config-object using `satisfies`.